### DISCUSSION TITLE
How to safely work with git branches

### DISCUSSION THREAD

**Discussion by unistar42** [community/community#108421] [Pull Requests] [answered]
### Select Topic Area

Question

### Body

I faced a problem with my git workflow.
I forked a fairly large project, made changes in the gh-pages branch, and then sent a merge request.
To start the work, I executed the git init command, then unpacked the project downloaded as a zip file and remote added it to my forked repository.
However, when I ran git status at this time, all files were updated, so when merging, the other side saw that I had modified all files.
So, I tried using the git clone command, but an error occurred during the clone process, so I cloned it with the git restore --source=HEAD :/ command, but even in this case, the entire file appeared to have been changed.
What should I do to avoid this? 
Is there a way to clone just one folder in a large project?

**Comment by sapalazz** [comment, marked as answer]
Hey there!

It sounds like you're running into a bit of a snag with your Git workflow, but no worries, we'll get it sorted out. Here are some simple steps to help you out:

### Cloning the repo
Instead of using git init and unpacking the zip file, directly clone the repository using git clone. This way, you'll get everything set up properly from the start.

```bash
git clone <repository-url>
```

### Cloning a specific folder
Git doesn't have a built-in way to clone just one folder, but you can still achieve that using a neat trick called sparse-checkout. After cloning, you can set up sparse-checkout to include only the folder you need. 

```bash
git clone <repository-url>
cd <repository-name>
git sparse-checkout init
git sparse-checkout set <folder-path>
```

Give these steps a try, they'll smooth out your workflow. Have a nice day. 😊👍

**Reply by hosseinmush** [reply]
gh repo clone moul/assh

**Comment by hosseinmush** [comment]
gh repo clone moul/assh

**Comment by hosseinmush** [comment]
<repository-url>
